marklogic-data-hub/src/main/resources/scaffolding/custom-module/sjs/main.sjs

```diff
@@ -82,17 +90,21 @@ function main(content, options) {
   //form our envelope here now, specifying our output format
   let envelope = datahub.flow.flowUtils.makeEnvelope(instance, headers, triples, outputFormat);
 
+  //create our return content object, we have a handy helper function for creating a json scaffolding, but you
+  //can also do a node-based one by using nodebuilder, especially if you're dealing with xml!
+  let newContent = datahub.flow.flowUtils.createContentAsObject();
```

Comment:
createContentAsObject returns what looks like an envelope, but then it's used to store "value" and "uri" and "context". So I'm not clear on the purpose of createContentAsObject.

Reply:
@aebadirad can you please enlighten us here?

Reply:
it creates a js object that make a full scaffolded 'content' object for someone to manipulate that the return of the step will turn into a document.

Reply:
It's created with triples/headers/instance properties, but those don't seem to be used - instead, value and uri are assigned. Or are those triples/headers/instance properties used somewhere?

Reply:
Hm, somehow createcontentasobject and createinstanceasobject didn't quite make it from the cherry picking.

Need to update those (content is supposed to be the raw content object)

Reply:
I was just trying to help get the 2735 change into 5.x-develop. I first cherry picked all the commits marked 2735 in develop and that did not work and was giving me NPE on step definition create. In 5.x-develop we should only have one custom module rather than having one for every step type since it was 5.1 requirement. @srinathgit and I discussed and agreed on that. Thats when this PR was created with just the mutability changes. At this point @aebadirad it will be quicker for you to either create another PR or push one commit to this one with the changes you mentioned.